python/hsfs/constructor/query.py

```diff
@@ -64,6 +64,12 @@ def _prep_read(self, online, read_options):
                 sql_query = query.query
             online_conn = None
 
+            if engine.get_instance().flyingduck_supported_and_enabled(
+                self, read_options
+            ):
+                read_options["use_flyingduck"] = True
+                return (self, sql_query), online_conn
+
```

Comment:
read_options["use_flyingduck"] is something I added initially only for internal use. I first check if the feature flag for flyingduck is set (arrow_flight_server.is_enabled()), and the check if the query is supported; if both conditions are met I set read_options["use_flyingduck"]=True so the python engine knows it should call flyingduck instead of hive. Only later I added a check to see if read_options["use_flyingduck"]  was already set by the user. The ise was that the use could overrule the feature flag and query flyignduck even if it is disabled (mainly for testing). But I reverted the changes now. Hope it's clearer now.